### ISSUE TITLE
Avoid set unique to '' (empty string)

### ISSUE THREAD

**Issue by allevo** [veny/GearmaNode#27]
When the client submits a new job with this library, the packet contains always the unique id.
If the developer doesn't set it, the '' (empty string) fallback is used. So when multiple tasks are sent with the same unique id (that is '', empty string), gearman server couldn't save correctly the queued jobs.
The problem begin here:
https://github.com/veny/GearmaNode/blob/master/lib/gearmanode/client.js#L147

Please consider this as a bug.

**Comment by veny** [comment]
hi Tommaso,
I do believe it is correct. see the documentation here: http://gearman.org/protocol/

you can see there an example of job submission:
```
Client job submission:

Client -> Job Server
00 52 45 51                \0REQ        (Magic)
00 00 00 07                7            (Packet type: SUBMIT_JOB)
00 00 00 0d                13           (Packet length)
72 65 76 65 72 73 65 00    reverse\0    (Function)
00                         \0           (Unique ID)
74 65 73 74                test         (Workload)
```

**Comment by allevo** [comment]
Used the following code (without a unique id):
```
'use strict';

var gearmanode = require('gearmanode');
var client = gearmanode.client();

var job = client.submitJob('reverse', 'hello world!', {background: true});
job.on('submitted', function() {
    console.log('AAAAA');
    client.close();
});
```
The sent buffer is this:
```
0 52 45 53
0 0 0 8
0 0 0 a
48 3a 61 6c 6c 65 76 69 3a 32
```

Setting the unique to 'a', the generated buffer is this:
```
0 52 45 53
0 0 0 8
0 0 0 a
48 3a 61 6c 6c 65 76 69 3a 33
```
As you can see, the buffet is the same to part of last letter.
Why?

**Comment by veny** [comment]
I disagree. there is my try:
```
var job = client.submitJob('reverse', 'hello world!');
>> debug: packet encoded, type=SUBMIT_JOB, buffer.size=33
>> verbose: SENT packet: 0 52 45 51 0 0 0 7 0 0 0 15 72 65 76 65 72 73 65 0 0 68 65 6c 6c 6f 20 77 6f 72 6c 64 21
```
and then with unique:
```
var job = client.submitJob('reverse', 'hello world!', {unique: 'Foo'});
>> debug: packet encoded, type=SUBMIT_JOB, buffer.size=36
>> verbose: SENT packet: 0 52 45 51 0 0 0 7 0 0 0 18 72 65 76 65 72 73 65 0 46 6f 6f 0 68 65 6c 6c 6f 20 77 6f 72 6c 64 21
```

I added following line on protocol.js#190:
```
logger.log('verbose', 'SENT packet: %s', common.bufferAsHex(buff));
```

**Comment by allevo** [comment]
Could you try with a background job?
As you underline, the packet that contains a unique, is larger then a packet that doesn't.
Probably the problem is present only with a background job. Tomorrow I will make more tests

**Comment by veny** [comment]
there is the reason: if you want to propagate the Unique from client to worker, you have to grab the job from gearman server with GRAB_JOB_UNIQ command.
unfortunately it was the last not implemented command.

I just released version 0.9.0 with fully implemented protocol. so if you initialize a worker like this:
```javascript
var worker = gearmanode.worker({withUnique: true});
```
and you send a job with Unique like this:
```javascript
client.submitJob('reverse', 'alfa', {unique: 'foo'});
```
than you can use `job.unique` attribute in worker.

**Comment by veny** [comment]
solved in v0.9.0
